src/DynamoPackages/PackageManagerExtension.cs

```diff
@@ -341,6 +347,36 @@ private void OnPackageRemoved(Package package)
         }
 
         #endregion
+        internal bool CheckIfPackagesTargetOtherHosts(IEnumerable<PackageVersion> newPackageHeaders)
+        {
+            // determine if any of the packages are targeting other hosts
+            var containsPackagesThatTargetOtherHosts = false;
+
+            // Known hosts
+            var knownHosts = PackageManagerClient.GetKnownHosts();
+
+            // Sandbox, special case: Warn if any package targets only one known host
+            if (String.IsNullOrEmpty(Host))
+            {
+                containsPackagesThatTargetOtherHosts =
+                    newPackageHeaders.Any(y => y.host_dependencies != null && y.host_dependencies.Intersect(knownHosts).Count() == 1);
```

Comment:
Why check only if the package targets one host? In other words, shouldn't this check be `... && y.host_dependencies.Intersect(knownHosts).Count() >= 1`

Reply:
this is a special case for sandbox - see the comment, because `host_dependencies` does not have a `core` or `sandbox` entry, the usual pattern is for authors to select *all the hosts* when they write a package for core. If a package only targets one host, there is a good chance this package does not work on sandbox.

Reply:
IMO `host_dependencies` need some redesign at some point in the future.